query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -379,57 +380,42 @@ private Filter createFilterForComplexAttributeExpressionFromOrderBy(
 
     List<Filter> childFilterList =
         orderByExpressionList.stream()
-            .filter(
-                orderByExpression ->
-                    orderByExpression.getExpression().getValueCase()
-                            == ValueCase.ATTRIBUTE_EXPRESSION
-                        && orderByExpression.getExpression().getAttributeExpression().hasSubpath())
-            .map(orderByExpression -> orderByExpression.getExpression().getAttributeExpression())
-            .map(
-                attributeExpression ->
-                    createContainsKeyFilter(
-                        attributeExpression.getAttributeId(), attributeExpression.getSubpath()))
+            .map(OrderByExpression::getExpression)
+            .filter(QueryRequestUtil::isComplexAttributeExpression)
+            .map(Expression::getAttributeExpression)
+            .map(QueryRequestUtil::createContainsKeyFilter)
             .collect(Collectors.toList());
 
     return childFilterList.isEmpty()
         ? Filter.getDefaultInstance()
         : Filter.newBuilder().setOperator(Operator.AND).addAllChildFilter(childFilterList).build();
```

Comment:
nit: If we care about optimizing the structure, we don't need to wrap a single child filter in an `AND` filter. Along the same lines, you could just return the list and further optimize in `rebuildFilterForComplexAttributeExpression` making the decision to `AND` there once we have gathered all the top level filters.

Reply:
Can be done. I guess whole of the code will remian similar bbut we will add a separate condition to check if childFilterList contains one or more filter. Should I do this ?

Reply:
added

Reply:
resolved

Reply:
Yeah, either doing it here or better yet, just return `childFilterList` and do that check in the caller once all of the top level filters are gathered.

Reply:
resolved